test/tasks/ods-package-image_test.go

```diff
@@ -2,20 +2,37 @@ package tasks
 
 import (
 	"fmt"
+	"net/url"
 	"os"
 	"path/filepath"
 	"strings"
 	"testing"
 
+	"github.com/google/go-cmp/cmp"
 	"github.com/opendevstack/pipeline/internal/command"
 	"github.com/opendevstack/pipeline/pkg/tasktesting"
 )
 
 func TestTaskODSPackageImage(t *testing.T) {
 	runTaskTestCases(t,
 		"ods-package-image",
-		[]tasktesting.Service{},
+		[]tasktesting.Service{
+			tasktesting.Nexus,
+		},
 		map[string]tasktesting.TestCase{
+			"task should build image and use nexus args": {
+				WorkspaceDirMapping: map[string]string{"source": "hello-nexus-app"},
+				PreRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+					ctxt.ODS = tasktesting.SetupGitRepo(t, ctxt.Namespace, wsDir)
+				},
+				WantRunSuccess: true,
+				PostRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+					checkResultingFiles(t, ctxt, wsDir)
+					checkResultingImageHelloNexus(t, ctxt, wsDir)
+				},
+			},
 			"task should build image": {
```

Comment:
This looks ace. Just wondering if we still get additional value from `task should build image` as a separate test? Maybe a little more clarity as to whether image building or the nexus integration is broken, but not sure if this warrants the extra time spent. What do you think?

Reply:
I believe https://github.com/opendevstack/ods-pipeline/blob/324ae359b476240b9d019b6256c92f58506b8576/test/testdata/workspaces/python-fastapi-sample-app/docker/Dockerfile will also be tested and would actually use this or?

Reply:
Final thing in my mind is to double check that the expected double quotes at https://github.com/opendevstack/ods-pipeline/blob/bf9a00e5a2ed00226877be76e9a534f885d4fe2d/test/tasks/ods-package-image_test.go#L165-L171
actually need to be there.

Reply:
@henrjk The python tests do not build an image as far as I am aware. See https://github.com/opendevstack/ods-pipeline/blob/master/test/tasks/ods-build-python_test.go. The tests are focused on the tasks alone right now. We might want to extend that with more e2e tests at some point ...

Hehe, I was also wondering about the double quotes but then thought it doesn't harm (but since everything is encoded it should probably also work without?)

Reply:
I tested in an existing project using the jenkins pipeline and there these variables are not quoted. I am removing the quotes now.
Adding more language e2e tests makes sense to me, but should be done in another PR I suppose.